### DISCUSSION TITLE
Can I trigger Modal/Dialog outside of React Comopnent like Toast?

### DISCUSSION THREAD

**Discussion by joshua7v** [adobe/react-spectrum#4239] [Q&A] [answered]
toast has a toastQueue which you can add/remove from anywhere
But the state of Modal/Dialog comes from hooks like `useOverlayTriggerState`, which I can't use outside of a React component I suppose?

Basically what I want is to eliminate `<ModalTrigger />`s and just use a global function to trigger Dialogs
is there a way to achieve that?

**Comment by snowystinger** [comment, marked as answer]
I think you want something like https://react-spectrum.adobe.com/react-spectrum/DialogContainer.html
However, there are still some reasons to keep the trigger in many cases. They help a button communicate to screen readers that they will open a popup or dialog, whether it's already open or not, and which one that button controls. Specifically, these aria roles
```
aria-haspopup
aria-expanded
aria-controls
```

**Reply by joshua7v** [reply]
Thanks,

Yeah, I do agree that sometimes we need the triggers, not only because the aria roles, but also to colocate the specific dialog related logic.
I'm using react-aria only without the spectrum design system, so I missed the DialogContainer docs I guess, will check that out.
Though I probably couldn't just use that, but I'm sure it would give me some insights.

**Reply by snowystinger** [reply]
Totally missed we have an example https://react-spectrum.adobe.com/react-aria/Modal.html#controlled-open-state
